packages/next/client/image.tsx

```diff
@@ -401,6 +402,20 @@ function imgixLoader({ root, src, width, quality }: LoaderProps): string {
   return `${root}${normalizeSrc(src)}${paramsString}`
 }
 
+function fastlyLoader({ root, src, width, quality }: LoaderProps): string {
+  const url = new URL(`${root}${normalizeSrc(src)}`)
+  url.searchParams.set('auto', 'webp')
+  url.searchParams.set('optimize', 'medium')
+  url.searchParams.set('width', width.toFixed(0))
+  if (quality) {
+    url.searchParams.set('quality', quality.toFixed(2))
+  }
+  if (typeof window !== 'undefined' && window.devicePixelRatio > 1) {
```

Comment:
This will break hydration and preloading

Reply:
Yeah I was afraid of that, I will remove. Any other suggestions? I think ultimately we'd have to extend support for `srcset` to do this properly